bundles/org.openhab.binding.bluetooth.bluegiga/src/main/java/org/openhab/binding/bluetooth/bluegiga/BlueGigaBluetoothDevice.java

```diff
@@ -193,10 +194,28 @@ public void bluegigaEventReceived(BlueGigaResponse event) {
                         case EIR_FLAGS:
                             break;
                         case EIR_MANUFACTURER_SPECIFIC:
-                            manufacturerData = (byte[]) eir.getRecord(EirDataType.EIR_MANUFACTURER_SPECIFIC);
-                            if (manufacturerData.length > 2) {
-                                int id = manufacturerData[0] + (manufacturerData[1] << 8);
-                                manufacturer = id;
+                            Object obj = eir.getRecord(EirDataType.EIR_MANUFACTURER_SPECIFIC);
+                            if (obj != null && obj instanceof Map<?, ?>) {
+                                try {
+                                    @SuppressWarnings("unchecked")
+                                    Map<Short, int[]> eirRecord = (Map<Short, int[]>) obj;
+                                    Map.Entry<Short, int[]> eirEntry = eirRecord.entrySet().iterator().next();
+
+                                    manufacturer = (int) eirEntry.getKey();
+
+                                    int[] manufacturerInt = eirEntry.getValue();
+                                    manufacturerData = new byte[manufacturerInt.length + 2];
+                                    // Convert short Company ID to bytes and add it to manufacturerData
+                                    manufacturerData[0] = (byte) (manufacturer & 0xff);
+                                    manufacturerData[1] = (byte) ((manufacturer >> 8) & 0xff);
+                                    // Add Convert int custom data nd add it to manufacturerData
+                                    for (int i = 0; i < manufacturerInt.length; i++) {
+                                        manufacturerData[i + 2] = (byte) manufacturerInt[i];
+                                    }
+                                } catch (ClassCastException e) {
+                                    logger.debug("Unsupported manufacturer specific record received from device {}",
+                                            address);
+                                }
```

Comment:
If you are going to be catching a ClassCastException you might as well forego the `instanceof` check.

Reply:
It doesn't disturb maintainability of the code (so it's not much extra boilerplate code) and I think in terms of performance it's a bit cheaper then the ClassCastException. Nevertheless, I have no trong opinion on this.

Reply:
The ideal solution would involve generics to discern the type for a EirRecord, but I think that makes things a bit more complicated than is necessary. At the very least, make the try/catch block as small as possible so it is easy to understand the cast that is "_expected_" to go wrong.

Reply:
I agree with generics, but I also agree it would exceed the scope of this PR.

Reply:
I decided for removing the instanceof check to not further segregate the code.